dotnet/src/SemanticKernel.Abstractions/Filters/FunctionCall/IFunctionCallFilter.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel;
+
+#pragma warning disable CA1716 // Identifiers should not match keywords (Func<FunctionCallInvocationContext, Task> next)
+
+/// <summary>
+/// Interface for filtering actions during function calling invocation.
+/// </summary>
+[Experimental("SKEXP0001")]
+public interface IFunctionCallFilter
```

Comment:
Wondering if the interface name and description are clear enough.

Based on naming it's not obvious what the difference between `IFunctionFilter` and `IFunctionCallFilter` is.

We are also using the term `ToolCallBehavior`.

Some thoughts:

- `IToolCallFilter`
- `IToolFunctionFilter`
- `IToolFunctionCallFilter`

Let's discuss with @matthewbolanos

Reply:
Agreed on following renaming:
- `IPromptFilter` -> `IPromptRenderFilter`
- `IFunctionFilter` -> `IFunctionInvocationFilter`
- `IFunctionCallFilter` -> `IAutoFunctionInvocationFilter`

Renaming for function and prompt filters will be completed in following PR.

Reply:
@dmytrostruk I may have not aware of the agreement, but if thats the case, ignore my comments. I don't like the `Invocation` mixed with the other. Those are different things and can be very confusing as Auto Function Invocation is as concept that only how already knows Semantic Kernel understand, `Function Calling` although is something that community has an understanding.

`AutoFunctionCallRequestFilter` would be my suggestion for this.

Reply:
@RogerBarreto based on discussion with @matthewbolanos we decided to proceed with this naming. I think it will work fine with `ToolCallBehavior.AutoInvokeKernelFunctions` setting.